### ISSUE TITLE
Undocumented breaking change in "valid-jsdoc" rule (Missing JSDoc @returns for function)

### ISSUE THREAD

**Issue by doberkofler** [eslint/eslint#10510]
**Tell us about your environment**

* **ESLint Version:** 5.0.0
* **Node Version:** 10.4.1
* **npm Version:** 6.1.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "parserOptions": {
        "ecmaVersion": 8,
        "sourceType": "script",
        "ecmaFeatures": {}
    },
    "rules": {
        "valid-jsdoc": 2
    },
    "env": {}
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
/**
 * my function
 */
async function test() {
	await new Promise.resolve(true);
}

test();
```

```bash
eslint *.js
```

**What did you expect to happen?**
In eslint 4.19.1 the jsdoc rule did not complain about a missing @returns.

**What actually happened? Please include the actual, raw output from ESLint.**
The error `Missing JSDoc @returns for function (valid-jsdoc)` is reported

**Comment by doberkofler** [comment]
At lest for me this is just a minor issue that should be part of migration documentation.

**Comment by Aladdin-ADD** [comment]
thanks for the report! seems a bug to me.

**Comment by ljharb** [comment]
It’s an async function - it has only one possible return, a promise, and it would be redundant to require documenting that.

**Comment by doberkofler** [comment]
@ljharb I personally think it can be helpful to also document the return and specify what type the promise will resolve to. I generally have no problem with specifying the return, but it should be documented and eventually made optional for people not interested.

**Comment by not-an-aardvark** [comment]
Are you sure the rule wasn't complaining about a missing `@returns` in v4.19.1? From testing it out locally it seems like a problem is also being reported when I lint this code with v4.19.1, but it's possible I'm missing something.

**Comment by doberkofler** [comment]
@not-an-aardvark Quite sure but not 100%. I noticed immediately after the just updating eslint to 5.0.0 but did not revert back and reinstall all packages.

**Comment by not-an-aardvark** [comment]
Actually, there seem to be a few relevant issues https://github.com/eslint/eslint/issues/9881 https://github.com/eslint/eslint/pull/10161 https://github.com/eslint/eslint/issues/10386 https://github.com/eslint/eslint/pull/10480

Specifically, I think this issue is a duplicate of https://github.com/eslint/eslint/issues/10386.
